zyte_api/aio/client.py

```diff
@@ -146,10 +146,19 @@ def request_parallel_as_completed(self,
         """
         sem = asyncio.Semaphore(self.n_conn)
 
+        if session is not None:
+            close_session = False
+        else:
+            session = create_session(n_conn=self.n_conn)
+            close_session = True
+
         async def _request(query):
             async with sem:
                 return await self.request_raw(query,
                     endpoint=endpoint,
                     session=session)
 
-        return asyncio.as_completed([_request(query) for query in queries])
+        yield from asyncio.as_completed([_request(query) for query in queries])
+
+        if close_session:
+            asyncio.run(session.close())
```

Comment:
Should it be in `finally:` section? For example, if a user code which uses this method raises an exception, is the session going to be closed?